### ISSUE TITLE
Feature request: rotated installation

### ISSUE THREAD

**Issue by proggprod** [Sian-Lee-SA/Home-Assistant-Switch-Manager#119]
Sometimes I install light switches like the Opple switches at 90 deg to have two up/down buttons instead of of left/right buttons. 

Can you add a button in the UI to rotate the switch blueprints?

**Comment by Sian-Lee-SA** [comment]
While this might be more "common" in switches with the design of the device you described, for 90% of other devices this wouldn't be and the added button/function would be an unnecessary item showing for those. For now you can simply just copy the blueprint and call it -vertical or -horizontal etc then rotate the image and reposition the buttons where necessary.

**Comment by proggprod** [comment]
I am coming from an openHAB setup where I was using something like this. I allowed some of the switch types I have to be rotated depending on the style. 90% of switches being unsuitable might be a high estimate, especially in my case where I have 90% of switches that _are_ suitable for rotating. It's cleaner and easier to manage if there is a rotate option instead of having multiple switch types (of course behind the scenes there are separate templates for each of the mappings to make sense semantically: left-up or top-right for example). 
Just because it doesn't apply to every switch doesn't make it a bad idea.

**Comment by Sian-Lee-SA** [comment]
It's an educated estimated based on my calculations eg. How many devices I've reviewed, you being the first to request such feature, AU and US wall plates are vertical which results in an upright position, labelling (whether be numbers, or symbols representing numbers) on many devices that only make sense to be placed in one way. But having seen the Opple, I also understand it was designed to be interchangeable with how it's installed so I understand your point and yes it's valid.

So I'm not dismissing the idea, and if or when there's more request like this then it will put it higher on my deliberation and priority list. I just want to be clear, Switch Manager in my vision was to be intuitive but not bombarded with features otherwise it get's away from simplicity. The idea is also to allow users that customisation within their own blueprints and the use of variables etc. to fill such gaps where Switch Manager didn't need micro management.

And no, it's not a bad idea it's just figuring the percentage of use and whether copied blueprints (as custom blueprints and not in the repo) is better suited.

**Comment by GCSabas** [comment]
Wanted to put in a very late +1 on this, as I DO think it would be helpful, but it's certainly of no great importance.


First, this repo has let me not only get a product abandoned by its manufacturer working again, but it's more reliable and can have additional features!  So, thanks a ton!
And I'm going to have to beg forgiveness if I messed up something in the way I forked this repo, but I wanted to still chime in on this, briefly.

EnOcean's rocker switches, whether they're the modern WiFi/BT versions or the older sub-GHz stuff tends to be square to allow for vertical or horizontal placement.
And in my case, reviving the Senic Nuimo Click devices, the typical install is horizontal.

To that end, I first made a generic EnOcean switch, then a rotated-90-degree version for the proper Nuimo Click device with labels.

https://github.com/SadFatFrog/Switch-Manager-EnOcean-Cynical/commit/ae813ca55c2088f489a834252c46e25280d77495

I have not written up the vertical version of the Senic device (as it's the same as the generic), but felt I'd at least pop the image up on my own for as I might use it at some point.



Anyway, I know the request is closed, but still wanted to toss this out there, just in case.

Thanks again!
